azure/components/computecluster/setup.ftl

```diff
@@ -326,12 +331,13 @@
     [@createVMScaleSetExtension
         id=extension.Id
         name=extension.Name
-        publisher=extConfig.Publisher!""
-        type=extConfig.Type.Name!""
-        typeHandlerVersion=extConfig.Type.HandlerVersion!""
-        autoUpgradeMinorVersion=extConfig.AutoUpgradeOnMinorVersion
+        publisher=extensionScriptConfig.Publisher
+        type=extensionScriptConfig.Type.Name
+        typeHandlerVersion=extensionScriptConfig.Type.HandlerVersion
+        autoUpgradeMinorVersion=extensionScriptConfig.AutoUpgradeOnMinorVersion!true
```

Comment:
Would it be better to pass in the extensionScriptConfig Object and the manage how to determine the publisher etc inside the createVMScaleSetExtension macro?

Reply:
Yep, [much tidier](
https://github.com/hamlet-io/engine-plugin-azure/pull/149/commits/8dae49df93b55cf388c8585fb74d0bab6a484781)